pkg/build/graph/edges.go

```diff
@@ -55,24 +57,37 @@ func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildCon
 		g.AddEdge(in, node, BuildInputEdgeKind)
 	}
 
+	sourceImages := []*kapi.ObjectReference{}
 	from := buildutil.GetImageStreamForStrategy(node.BuildConfig.Spec.Strategy)
 	if from != nil {
-		switch from.Kind {
+		sourceImages = append(sourceImages, from)
```

Comment:
from is only a sourceImage if an ICT exists with no from, right?  is that pre-condition guaranteed here?  if not it should move down into the loop below.